pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -62,13 +61,18 @@
 import java.util.TimeZone;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 
 /**
  * QueryExecutor implementation for the V3 protocol.
  */
 public class QueryExecutorImpl extends QueryExecutorBase {
 
   private static final Logger LOGGER = Logger.getLogger(QueryExecutorImpl.class.getName());
+  private static final Pattern COMMAND_COMPLETE_PATTERN = Pattern.compile("^([A-Za-z]+)(?: (\\d+))?(?: (\\d+))?$");
```

Comment:
I suggest to use `"([A-Za-z]++)(?: (\\d++))?+(?: (\\d++))?+"`
It is explained in https://www.regular-expressions.info/catastrophic.html

Reply:
For normal/valid responses from the backend server I don't think there's any backtracking as it'd be either COMMAND, COMMAND COUNT, or COMMAND OID COUNT so the internal FSA would terminate. But if the regex can be improved to deal with a malicious response without too much complexity then sure why not. I'll take a look.

Reply:
Let's put it in another way: `++` forbids backtracking, and it makes the code DOS-safe.